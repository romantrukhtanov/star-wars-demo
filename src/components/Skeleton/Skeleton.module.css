.root {
  background-color: #696969;
}

.rounded {
  border-radius: 8px;
}
