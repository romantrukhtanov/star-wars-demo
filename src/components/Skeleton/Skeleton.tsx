import type { SkeletonProps } from '@mui/material';
import { Skeleton as MUISkeleton } from '@mui/material';

import styles from './Skeleton.module.css';

type Props = Omit<SkeletonProps, 'classes'>;

const classes: SkeletonProps['classes'] = {
  root: styles.root,
  rounded: styles.rounded,
};

export function Skeleton({ variant, height = '100%', ...rest }: Props) {
  return (
    <MUISkeleton
      classes={classes}
      animation="wave"
      variant={variant ?? 'rounded'}
      height={height}
      {...rest}
    />
  );
}
