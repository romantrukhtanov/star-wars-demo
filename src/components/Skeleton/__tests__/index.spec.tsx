import { describe, expect, it } from 'vitest';

import { render } from '@/tests/test-utils';

import { Skeleton } from '../Skeleton';

describe('Skeleton', () => {
  it('render component', () => {
    const { baseElement } = render(<Skeleton />);
    expect(baseElement).toMatchSnapshot();
  });
});
