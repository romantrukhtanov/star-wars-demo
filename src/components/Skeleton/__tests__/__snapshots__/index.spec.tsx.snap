// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Skeleton > render component 1`] = `
<body>
  <div>
    <span
      class="MuiSkeleton-root _root_2cc9eb MuiSkeleton-rounded _rounded_2cc9eb MuiSkeleton-wave css-1hirv1y-MuiSkeleton-root"
      style="height: 100%;"
    />
  </div>
</body>
`;
